lib/Subscriber/Core/MappingSubscriber.php

```diff
@@ -111,14 +116,21 @@ private function persistReference(NodeInterface $node, DocumentAccessor $accesso
         }
 
         if ($fieldMapping['multiple']) {
-            throw new \InvalidArgumentException(sprintf(
-                'Mapping references as multiple not currently supported (when mapping "%s"',
-                $fieldName
-            ));
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Mapping references as multiple not currently supported (when mapping "%s")',
+                    $fieldName
+                )
+            );
         }
 
         $referenceNode = $this->documentRegistry->getNodeForDocument($referenceDocument);
-        $phpcrName = $this->encoder->encode($fieldMapping['encoding'], $fieldMapping['property'], $locale);
+        try {
+            $phpcrName = $this->encoder->encode($fieldMapping['encoding'], $fieldMapping['property'], $locale);
+        } catch (\InvalidArgumentException $ex) {
+            // arguments unvalid no valid propertyname could be generated (e.g. no locale given for localized encoding)
+            return;
+        }
         $node->setProperty($phpcrName, $referenceNode);
```

Comment:
Why don't you just move that also into the try block, and therefore remove the `return` statement in the catch block? I think we said we either return early or at the very end, that's somewhere in between.